edgelet/edgelet-http-mgmt/src/server/module/prepare_update.rs

```diff
@@ -49,13 +50,35 @@ where
             })
             .and_then(|(core_spec, runtime)| {
                 let name = core_spec.name().to_string();
-                runtime.registry().pull(core_spec.config()).then(|result| {
-                    result.with_context(|_| ErrorKind::PrepareUpdateModule(name.clone()))?;
-                    Ok(name)
-                })
+                let image_pull_policy = *core_spec.image_pull_policy();
+                match image_pull_policy {
+                    ImagePullPolicy::OnCreate => Either::A(
+                        runtime
+                            .registry()
+                            .pull(core_spec.config())
+                            .then(move |result| {
+                                result.with_context(|_| {
+                                    ErrorKind::PrepareUpdateModule(name.clone())
+                                })?;
+                                Ok((name, image_pull_policy))
+                            }),
+                    ),
+                    ImagePullPolicy::Never => {
+                        Either::B(futures::future::ok((name, image_pull_policy)))
+                    }
+                }
             })
-            .and_then(|name| -> Result<_, Error> {
-                debug!("Successfully pulled new image for module {}", name);
+            .and_then(|(name, image_pull_policy)| -> Result<_, Error> {
+                match image_pull_policy {
+                    ImagePullPolicy::OnCreate => {
```

Comment:
Nit: This will work, but instead of adding another check for image_pull_policy, we can just pass a flag to indicate if the image was pulled or not.

Reply:
Yep, will change it to just use a boolean flag